src/webgpu/compat/api/validation/pipeline_creation.spec.ts

```diff
@@ -0,0 +1,146 @@
+          textureType: 'texture_depth_cube',
+        },
+        {
+          sampleWGSL: 'textureSample(t, s, vec2f(0), 0)',
+          textureType: 'texture_depth_2d_array',
+        },
+        {
+          sampleWGSL: 'textureSample(t, s, vec2f(0), vec2i(0, 0))',
+          textureType: 'texture_depth_2d',
+        },
+        {
+          sampleWGSL: 'textureSample(t, s, vec2f(0), 0, vec2i(0, 0))',
+          textureType: 'texture_depth_2d_array',
+        },
+        {
+          sampleWGSL: 'textureSampleLevel(t, s, vec2f(0), 0)',
+          textureType: 'texture_depth_2d',
+        },
+        {
+          sampleWGSL: 'textureSampleLevel(t, s, vec3f(0), 0)',
+          textureType: 'texture_depth_cube',
+        },
+        {
+          sampleWGSL: 'textureSampleLevel(t, s, vec2f(0), 0, 0)',
+          textureType: 'texture_depth_2d_array',
+        },
+        {
+          sampleWGSL: 'textureSampleLevel(t, s, vec2f(0), 0, vec2i(0, 0))',
+          textureType: 'texture_depth_2d',
+        },
+        {
+          sampleWGSL: 'textureSampleLevel(t, s, vec2f(0), 0, 0, vec2i(0, 0))',
+          textureType: 'texture_depth_2d_array',
+        },
+        {
+          sampleWGSL: 'textureGather(t, s, vec2f(0))',
+          textureType: 'texture_depth_2d',
+        },
+        {
+          sampleWGSL: 'textureGather(t, s, vec3f(0))',
+          textureType: 'texture_depth_cube',
+        },
+        {
+          sampleWGSL: 'textureGather(t, s, vec2f(0), 0)',
+          textureType: 'texture_depth_2d_array',
+        },
+        {
+          sampleWGSL: 'textureGather(t, s, vec2f(0), vec2i(0, 0))',
+          textureType: 'texture_depth_2d',
+        },
+        {
+          sampleWGSL: 'textureGather(t, s, vec2f(0), 0, vec2i(0, 0))',
+          textureType: 'texture_depth_2d_array',
+        },
+      ])
+      .combine('stage', kShaderStages)
+      .filter(t => t.sampleWGSL.startsWith('textureGather') || t.stage === 'fragment')
+      .combine('async', [false, true] as const)
+  )
+  .fn(t => {
+    const { sampleWGSL, textureType, stage, async } = t.params;
+
+    const usageWGSL = `_ = ${sampleWGSL};`;
+    const module = t.device.createShaderModule({
+      code: `
+        @group(0) @binding(0) var t: ${textureType};
+        @group(1) @binding(0) var s: sampler;
+
+        // make sure it's fine such a combination exists but it's not used.
+        fn unused() {
+          ${usageWGSL};
+        }
+
+        @vertex fn vs() -> @builtin(position) vec4f {
+            ${stage === 'vertex' ? usageWGSL : ''}
+            return vec4f(0);
+        }
+
+        @fragment fn fs() -> @location(0) vec4f {
+            ${stage === 'fragment' ? usageWGSL : ''}
+            return vec4f(0);
+        }
+
+        @compute @workgroup_size(1) fn cs() {
+            ${stage === 'compute' ? usageWGSL : ''};
+        }
+      `,
+    });
+
+    const success = !t.isCompatibility || textureType === 'texture_2d<f32>';
```

Comment:
nit: this is under `compat/` so I assume `!t.isCompatibility` is not needed?

Reply:
It was decided all of the compat tests should run in core and pass in the case where compat fails, as a validation of the test itself. In other words, the test is testing that something that's allowed in core is not allowed in compat. Checking it passes in core checks the test is correct.